### ISSUE TITLE
Mono audio from MICs?

### ISSUE THREAD

**Issue by adrianrb** [hamuro80/blackstomp#1]
First, THANK YOU for your effort.
But, my audio is mono...
Can you help me?
(yes, i´m using ESP32-A1S kit)

**Comment by hamuro80** [comment]
Yes it works with ESP32-A1S kit. To be simple, you can process only the left or right channel, then use only left or right output. To be complicated, you can edit the codec and esp's i2s setting to work in mono mode.

**Comment by adrianrb** [comment]
Hi, me again.
Do you have/know the A1S Kit?
Another issue (I think maybe with my board), is there no input in LINEIN...
Tested with your code (arduino) *AND* expressif example (passthrough).
Any thoughts?

**Comment by adrianrb** [comment]
Yes, Im´m trying to build a guitar fx "stomp"...
Many ideias... ;-)

**Comment by hamuro80** [comment]
I'm sorry I thought it was ESP32-A1S Audio Kit from Ai-Thinker (which has line input and stereo microphone). Is there any link for the A1S Kit?

**Comment by adrianrb** [comment]
Yes, IT IS... Ai-Tinker ESP32-A1S Audio Kit...
[this ](https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.exp-tech.de%2Fen%2Fplatforms%2Fesp32%2F9181%2Fesp32-a1s-wi-fi-bt-audio-development-kit&psig=AOvVaw30ugEeoXZU9VLchB9c-Woe&ust=1581785492197000&source=images&cd=vfe&ved=0CAIQjRxqFwoTCKCk9MXA0ecCFQAAAAAdAAAAABAD)board (v2.2 - only 2 mics, not 4)

**Comment by hamuro80** [comment]
I didn't say it has 4 microphone, it works.
